compiler/rustc_mir_transform/src/const_prop.rs

```diff
@@ -118,8 +118,8 @@ impl<'tcx> MirPass<'tcx> for ConstProp {
 
         // Traverse the body in reverse post-order, to ensure that `FullConstProp` locals are
         // assigned before being read.
-        let postorder = body.basic_blocks.postorder().to_vec();
-        for bb in postorder.into_iter().rev() {
+        let rpo = body.basic_blocks.reverse_postorder().to_vec();
+        for bb in rpo {
```

Comment:
@tmiasko This is where it could have been cool to avoid this existing allocation, as it's not easy to combine lazy RPO + `as_mut_preserves_cfg`. Probably not worth it just for this single call-site though, I don't remember seeing this elsewhere.